exec/java-exec/src/main/java/org/apache/drill/exec/rpc/security/ServerAuthenticationHandler.java

```diff
@@ -208,8 +207,12 @@ void process(SaslResponseContext<S, T> context) throws Exception {
 
         handleSuccess(context, challenge, saslServer);
       } else {
-        logger.info("Failed to authenticate client from {}", context.remoteAddress);
-        throw new SaslException("Client allegedly succeeded authentication, but server did not. Suspicious?");
+        final S connection = context.connection;
+        logger.info("Failed to authenticate client from {} with encryption context:{}",
+          connection.getRemoteAddress().toString(),
```

Comment:
getEncryptionString is returning String literal and that will be cache by JVM in string pool after first creation.